invoice/src/main/java/org/killbill/billing/invoice/api/user/DefaultInvoiceUserApi.java

```diff
@@ -269,23 +269,32 @@ public Invoice getInvoiceByInvoiceItem(final UUID invoiceItemId, final TenantCon
     public Invoice triggerDryRunInvoiceGeneration(final UUID accountId, final LocalDate targetDate, final DryRunArguments dryRunArguments, final CallContext context) throws InvoiceApiException {
         final InternalCallContext internalContext = internalCallContextFactory.createInternalCallContext(accountId, context);
 
-        final Invoice result = dispatcher.processAccount(true, accountId, targetDate, dryRunArguments, false, internalContext);
-        if (result == null) {
+        final Iterable<Invoice> results = dispatcher.processAccount(true, accountId, targetDate, dryRunArguments, false, internalContext);
+        if (results == null) {
             throw new InvoiceApiException(ErrorCode.INVOICE_NOTHING_TO_DO, accountId, targetDate != null ? targetDate : "null");
         } else {
-            return result;
+            if (Iterables.<Invoice>size(results) != 1) {
+                // If we change the API one day, we need to fix the implementation (see comments in processAccountWithLockAndInputTargetDate)
+                throw new UnsupportedOperationException("Invoices were generated but this version of Kill Bill doesn't support returning grouped invoices");
+            } else {
+                return Iterables.<Invoice>getOnlyElement(results);
+            }
         }
     }
 
     @Override
     public Invoice triggerInvoiceGeneration(final UUID accountId, @Nullable final LocalDate targetDate, final CallContext context) throws InvoiceApiException {
         final InternalCallContext internalContext = internalCallContextFactory.createInternalCallContext(accountId, context);
 
-        final Invoice result = dispatcher.processAccount(true, accountId, targetDate, null, false, internalContext);
-        if (result == null) {
+        final Iterable<Invoice> results = dispatcher.processAccount(true, accountId, targetDate, null, false, internalContext);
+        if (results == null) {
             throw new InvoiceApiException(ErrorCode.INVOICE_NOTHING_TO_DO, accountId, targetDate != null ? targetDate : "null");
         } else {
-            return result;
+            if (Iterables.<Invoice>size(results) != 1) {
+                throw new UnsupportedOperationException("Invoices were generated but this version of Kill Bill doesn't support returning grouped invoices");
```

Comment:
This one is trickier: Perhaps we could return one invoice with a `null` `invoiceId` which means that several invoices were created as part of grouping feature. Unexposed functionality...

Doing this would allow to write a custom endpoint from plugin that would call this api and then re-apply its grouping logic?

Reply:
This might break Kaui though: the `Location` header wouldn't be able to be created.

Let's chat IRL about both use-cases, on one hand, I would prefer to be explicit. On the other, I agree that it's a huge limitation to block these calls.